test/javascripts/components/chat-user-displayname-test.js

```diff
@@ -0,0 +1,76 @@
+import componentTest, {
+  setupRenderingTest,
+} from "discourse/tests/helpers/component-test";
+import { discourseModule, query } from "discourse/tests/helpers/qunit-helpers";
+import hbs from "htmlbars-inline-precompile";
+import { set } from "@ember/object";
+
+function displayname() {
+  return query(".chat-user-displayname").innerText.trim();
+}
+
+discourseModule(
+  "Discourse Chat | Component | chat-user-displayname | prioritize username in UX",
+  function (hooks) {
+    setupRenderingTest(hooks);
+
+    componentTest("username and no name", {
+      template: hbs`{{chat-user-displayname user=user}}`,
+
+      async beforeEach() {
+        set(this.siteSettings, "prioritize_username_in_ux", true);
```

Comment:
`this.siteSettings.set()` doesn't work there?

Reply:
I don't know, haven't tried for a long time, I got used to follow this pattern for whatever reason, will check later :+1:

Reply:
We talked of this with cvx and apparently during the tests siteSettings becomes a POJO making this not possible, we should fix it in core.